duckduckgo/duckduckgo-privacy-extension
# Refactor element hiding to make it more dynamic and less cpu-intensive

**Reviewer:** @kdzwinel 

**CC:** @jdorweiler @russellholt 

## Description:
This updates the element hiding data flow so that blocked frames are now dealt with as requests are blocked. Previously a list of blocked requests was fetched once per frame at the time of load, however that encountered issues when trackers were written into iframes after page load via document.write calls. Additionally, this now properly handles iframes that don't have src attributes.

## Steps to test this PR:
1. Test on yahoo.com to ensure no regressions in functionality have occurred.
2. Test on engadget.com to ensure that blocked ads are hidden properly (there's usually a blocked ad in the top right of the page that should be collapsed).
3. Whitelist one of these domains, and reload page. Element hiding should be disabled.
3. Test on non-oath domain to check that element hiding is disabled there.


## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
